sqlmesh/core/test/definition.py

```diff
@@ -177,7 +184,7 @@ def assert_equal(
         """Compare two DataFrames"""
         if partial:
             intersection = actual[actual.columns.intersection(expected.columns)]
-            if not intersection.empty:
+            if len(intersection.columns) > 0:
```

Comment:
if intersection.columns

Reply:
Pandas doesn't like treating `.columns` as truthy:

```
>>> import pandas as pd
>>> temp = pd.DataFrame({})
>>> if temp.columns:
...     print("truthy")
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/trey/tobiko/sqlmesh/.venv/lib/python3.11/site-packages/pandas/core/indexes/base.py", line 3170, in __nonzero__
    raise ValueError(
ValueError: The truth value of a RangeIndex is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
```

Should I use `intersection.columns.any()` instead?

Reply:
nah this is fine, didn't realize it was pandas